### ISSUE TITLE
Slack alert noise (meta)

### ISSUE THREAD

**Issue by rossfisherkc** [kubecost/cost-analyzer-helm-chart#2198]
KC Version 1.102 (when these alerts were introduced)

1. https://kubecost.zendesk.com/agent/tickets/3873
<img width="574" alt="image" src="https://user-images.githubusercontent.com/113053906/235224195-a78be375-8f0c-4a3a-a729-ef67d89c94b5.png">
<img width="468" alt="image" src="https://user-images.githubusercontent.com/113053906/235224220-b2e718b6-56c3-48a9-8d79-89176e6e5e97.png">

ETLs and bug reports look healthy. Please see thread for details and screenshots.

2. https://kubecost.zendesk.com/agent/tickets/3966
<img width="110" alt="image" src="https://user-images.githubusercontent.com/113053906/235224525-b02929f6-a654-46e8-811f-f1ae563c343b.png">

Clusters added noise

Will add more as they come through

**Comment by kwombach12** [comment]
Thanks @rossfisherkc any sense of what's causing all the noise? Are these newly added clusters which are throwing repeated alerts?

**Comment by wamak9** [comment]
Happening on ourside as well. Started getting these alerts every couple of hours.

```
[Kubecost Data Status Update]
You are receiving this alert because we have detected anomalies in your data feed to Kubecost.

Data Type: asset
Data Resolution: daily
Detected 'error' statuses for metric kubecost_asset_data_status
Data Type: asset
Data Resolution: hourly

Detected 'error' statuses for metric kubecost_asset_data_status```
To further investigate failure, visit Kubecost url:
```

![image](https://user-images.githubusercontent.com/15903405/235235518-7457cb7c-0aaa-488f-ad4a-a2c34ae1a278.png)

**Comment by keithhand** [comment]
I recently had another customer mention they see these same ETL error alerts, and they don't seem to have missing ETLs aside from the current ones delayed because of Thanos

![image](https://user-images.githubusercontent.com/43491549/235239626-e5854c91-fcf3-4db0-a63c-40450ac42428.png)

[Ticket for reference](https://kubecost.zendesk.com/agent/tickets/3884)

**Comment by dwbrown2** [comment]
Marked as P0 for this (103) release.

**Comment by avrodrigues5** [comment]
For the first part related to ETL notifications:

Few follow up questions @rossfisherkc : 

1. What is the state of ETL when you likely suspected this ETL notification causing noise?  So in this case asset hourly , Can i have diagnostic page snapshot like Keith.'s of asset hourly or daily
2. Is it possible to port-forward prometheus and check kubecost_asset_status graph for status error & empty? If assetSet was well and truly empty at that time, its actually the notification working.

Also i see that the snapshot of cluster added/removed showing repeated. Did ETL notification also come up as noise? I dont have access to those zendesk ticket link to see the snapshot hence asking!

Cluster added removed slack notifications are unrelated to ETL data slack notifications, likely should be part of another investigation altogether.

-regards
Alan

**Comment by avrodrigues5** [comment]
The ETL data notification functionality lets the user know there is an error, warning and empty at the current state.
When a notification occurs, if you hit /etl/status endpoint, there will be a corresponding empty/warning/error.

One reason why the notification comes and when the user looks, we don't have a corresponding error because there's an automatic rebuild scheduled for allocation and asset that could repair the store that was in an error state; hence user might get notified, but when they look at  /etl/status endpoint after automated repair, he likely won't have the corresponding issue.     

Things to look for in the log that lets you know there was an automated repair of errors allocation asset are:

`ETL: Allocation: starting automated repair with of` -> Allocation repair
`ETL: Asset: starting automated repair with of ` -> Asset repair

CC: @AjayTripathy

**Comment by dwbrown2** [comment]
@AjayTripathy @avrodrigues5 @kwombach12 is this the noise in this alert that we were worried about or something else?

**Comment by kwombach12** [comment]
@dwbrown2 I synced with @avrodrigues5 - updates below:
1. The user got a number of Asset ETL errors for daily and hourly store. We sent a notification about these errors, and the logs show ETL went through its self-healing process which resolved all of the errors which is why ETL & bug reports looked healthy
2. We believe @AjayTripathy is looking at this issue - @AjayTripathy can you confirm?

**Comment by rockerrock137** [comment]
@kwombach12 please see: https://kubecost.zendesk.com/agent/tickets/3966

Ajay has worked with customer, details in case

```
[Ajay Tripathy]
Monday 11:35
@Daniel Naguri are these continuing? I actually don't see any repetition in the "cluster added" section. We'd expect these to fire once when each kubecost starts up. Is that not helpful for you?


Monday 20:39
yes these are continuing. I see continued repeated notifications of cluster added. May it sends a notification on every cost-model etl run ?

Avatar
[Ajay Tripathy]
Yesterday 09:42
I don't think so. It would only send on restart. Are there a lot of cost-model restarts?

Yesterday 11:05
No, its irrespective of restarts for all clusters. Many of those clusters have 0 restarts
```

**Comment by kwombach12** [comment]
@AjayTripathy do we have anyone looking at this?

**Comment by keithhand** [comment]
Another customer mentioned this regarding the fact that enabling this one alert type can also cause noise based on the number of alerts that get enabled. For instance, a user might be interested in the ETL health alerts but not alerts like when clusters get added. I believe this alert intends to have minimal configuration, but because this alert type encompasses a lot, does it make sense to have the option to disable specific alert types within this alert?

**Comment by kwombach12** [comment]
@keithhand I could see that making sense - I still think we have some bug with the cluster created alert firing too frequently. @teevans know if anyone has looked into this?

**Comment by rockerrock137** [comment]
> Another customer mentioned this regarding the fact that enabling this one alert type can also cause noise based on the number of alerts that get enabled. For instance, a user might be interested in the ETL health alerts but not alerts like when clusters get added. I believe this alert intends to have minimal configuration, but because this alert type encompasses a lot, does it make sense to have the option to disable specific alert types within this alert?

I like this, UI toggles per alert type

**Comment by keithhand** [comment]
A bug report and more screenshots are also in this [Zendesk ticket](https://kubecost.zendesk.com/agent/tickets/3884).

**Comment by dwbrown2** [comment]
I'd really like to keep this alert as simple as possible. It really shouldn't be noisy in any scenario I've seen. There's always the ability to setup custom alerts with Prom metrics. 

@rockerrock137 @keithhand do we know why you're getting more alerts than expected?

**Comment by keithhand** [comment]
@dwbrown2, for the case that I posted, it might be related to the underlying metrics related to the Thanos data not being synced yet. We generally expect the last 3 hours of ETL data to be empty, as well as the current day, and that seems to be the only thing I noticed [here](https://kubecost.zendesk.com/agent/tickets/3884). 

With regards to the comment about customizing which alerts come through, this was because activating the Kubecost health alert enables various different alerts, which I originally classified as "noise." Perhaps it's best to open another ticket for a Feature Request for that feature and leave this ticket for solving the underlying bug regarding the failed ETL/Added clusters.
